File: edupilot_api/EduPilot.Api/DTOs/SupervisorInfoDTO.cs
namespace EduPilot.Api.DTOs
{
    public class SupervisorInfoDTO
    {
        public List<SupervisorStudentsInfoDTO>? Students { get; set; }
        public Guid? InstitutionId { get; set; }
        public string? InstitutionName { get; set; }
    }

    public class SupervisorStudentsInfoDTO
    {
        public Guid StudentId { get; set; }
        public string StudentFirstName { get; set; }
        public string? StudentMiddleName { get; set; }
        public string StudentLastName { get; set; }
        public int StudentGrade { get; set; }
    }
}

File: edupilot/lib/models/dtos/supervisor_info_dto.dart
import 'package:edupilot/models/dtos/supervisor_students_info_dto.dart';

class SupervisorInfoDTO {
  final List<SupervisorStudentsInfoDTO>? students;
  final String? institutionId;
  final String? institutionName;

  const SupervisorInfoDTO({
    this.students,
    this.institutionId,
    this.institutionName
  });

  factory SupervisorInfoDTO.fromJson(Map<String, dynamic> json) => SupervisorInfoDTO(
        students: (json['students'] as List<dynamic>?)
            ?.map((e) => SupervisorStudentsInfoDTO.fromJson(e as Map<String, dynamic>))
            .toList(),
        institutionId: json['institutionId'] as String?,
        institutionName: json['institutionName'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'students': students?.map((e) => e.toJson()).toList(),
        'institutionId': institutionId,
        'institutionName': institutionName,
      };
}